src/pages/settings/Wallet/ExpensifyCardPage.tsx

```diff
@@ -161,6 +146,17 @@ function ExpensifyCardPage({
         GetPhysicalCardUtils.goToNextPhysicalCardRoute(domain, GetPhysicalCardUtils.getUpdatedPrivatePersonalDetails(updatedDraftValues, privatePersonalDetails));
     };
 
+    const sendValidateCode = () => {
+        const primaryLogin = account?.primaryLogin ?? '';
+        const loginData = loginList?.[primaryLogin];
+
+        if (loginData?.validateCodeSent) {
+            return;
+        }
+
+        requestValidateCodeAction();
```

Comment:
We're having 2 ways to receive magic code `REQUEST_CONTACT_METHOD_VALIDATE_CODE` and  `WRITE_COMMANDS.RESEND_VALIDATE_CODE`, in `requestValidateCodeAction`function, we don't add any data to `loginData`, if we keep this condition here that means magic code from `WRITE_COMMANDS.RESEND_VALIDATE_CODE` can be used with the magic return from `REQUEST_CONTACT_METHOD_VALIDATE_CODE`. Did you check this case? 😄

Reply:
Related bug:
https://github.com/Expensify/App/pull/49445#issuecomment-2406832833